tests/Microsoft.Bot.Builder.LanguageGeneration.Tests/Examples/RecursiveTemplate.lg

```diff
@@ -0,0 +1,20 @@
+﻿> 1 + 2 + 3 +...+ number
+# RecursiveAccumulate(number)
+- IF:${number <= 1}
+ - ${1}
+- ELSE:
+ - ${RecursiveAccumulate(number - 1) + number}
+
+> 1 * 2 * 3 * 4 * 5 * ... * number
+# RecursiveFactorial(number)
+- IF: ${number <= 1}
+ - ${1}
+- ELSE:
+ - ${RecursiveFactorial(number - 1) * number}
+
+> 1, 1, 2, 3, 5, 8, 13 ... number th
+# RecursiveFibonacciSequence(number)
+- IF: ${number <= 2}
+ - ${1}
+- ELSE:
+ - ${RecursiveFibonacciSequence(number - 1) + RecursiveFibonacciSequence(number - 2)}
```

Comment:
Is there already a test which should trigger the recursive evaluation?

Reply:
no, LG do not totally support recursive evaluation before.

Reply:
I meant a test which should trigger a recursive error.  These tests will succeed which is great, but where is the test that still causes the error, i.e. 
# shouldFail(x)
- ${shouldFail(x)}

---
In reply to: [434410130](https://github.com/microsoft/botbuilder-dotnet/pull/4012#discussion_r434410130) [](ancestors = 434410130)

Reply:
sure #Resolved